pkg/snap/snap/snapcraft.yaml

```diff
@@ -0,0 +1,76 @@
+name: solvespace
+base: core18
+summary: Parametric 2d/3d CAD
+adopt-info: solvespace
+description: |
+  SOLVESPACE is a free (GPLv3) parametric 3d CAD tool.
+  Applications include
+  * modeling 3d parts — draw with extrudes, revolves, and Boolean (union / difference) operations
+  * modeling 2d parts — draw the part as a single section, and export DXF, PDF, SVG; use 3d assembly to verify fit
+  * 3d-printed parts — export the STL or other triangle mesh expected by most 3d printers
+  * preparing CAM data — export 2d vector art for a waterjet machine or laser cutter; or generate STEP or STL, for import into third-party CAM software for machining
+  * mechanism design — use the constraint solver to simulate planar or spatial linkages, with pin, ball, or slide joints
+  * plane and solid geometry — replace hand-solved trigonometry and spreadsheets with a live dimensioned drawing
+
+grade: stable
+confinement: strict
+
+layout:
+  /usr/share/solvespace:
+    bind: $SNAP/usr/share/solvespace
+
+apps:
+  solvespace:
+    command: usr/bin/solvespace
+    extensions: [gnome-3-28]
+    plugs: [opengl, unity7, home, removable-media, gsettings, network]
+    desktop: usr/share/applications/solvespace.desktop
+    environment:
+      __EGL_VENDOR_LIBRARY_DIRS: $SNAP/usr/share/glvnd/egl_vendor.d
+  solvespace-cli:
+    command: usr/bin/solvespace-cli
+    plugs: [home, removable-media, network]
```

Comment:
Does the CLI actually work with strict confinement? As in, have you tried specifying paths?

Reply:
That works:

```bash
solvespace.solvespace-cli export-mesh -o test.stl -t 0.01 Desktop/testsketch.slvs 
Missing (absent) translation for group-name'#references'
Missing (absent) translation for group-name'sketch-in-plane'
Written 'test.stl'.
```

So path-wise that is OK. However I've got me some segfaults when trying to use the export-view command. I have to read a little about how to properly use gdb in snaps...

Reply:
What about absolute paths?

Reply:
Something a bit more exciting:

```bash
mfederle@maxi:~$ solvespace.solvespace-cli export-mesh -o /run/user/1095800001/gvfs/sftp:host=*****,user=pi/home/pi/test.stl -t 0.01 /home/mfederle/Desktop/testsketch.slvs 
Missing (absent) translation for group-name'#references'
Missing (absent) translation for group-name'sketch-in-plane'
Written '/run/user/1095800001/gvfs/sftp:host=hermes.internal.linergy.de,user=pi/home/pi/test.stl'

mfederle@maxi:~$ stat /run/user/1095800001/gvfs/sftp:host=*****,user=pi/home/pi/test.stl
  File: /run/user/1095800001/gvfs/sftp:host=hermes.internal.linergy.de,user=pi/home/pi/test.stl
  Size: 684       	Blocks: 2          IO Block: 4096   regular file
Device: 46h/70d	Inode: 16851       Links: 1
Access: (0644/-rw-r--r--)  Uid: (1095800001/mfederle)   Gid: (1095800001/mfederle)
Access: 2019-09-10 15:01:08.000000000 +0200
Modify: 2019-09-10 15:01:08.000000000 +0200
Change: 2019-09-10 15:01:08.000000000 +0200
 Birth: -
```

Something noteworthy, for the forbidden path ```/tmp/test.stl``` we get this:

```bash
mfederle@maxi:~$ solvespace.solvespace-cli export-mesh -o /tmp/test.stl -t 0.01 /home/mfederle/Desktop/testsketch.slvs 
Missing (absent) translation for group-name'#references'
Missing (absent) translation for group-name'sketch-in-plane'
Written '/tmp/test.stl'.

mfederle@maxi:~$ stat /tmp/test.stl
stat: cannot stat '/tmp/test.stl': No such file or directory
```

Reply:
OK, I have no idea *how* it works (it seems impossible to me! clearly, I don't know something important about Linux containers) but I'm fine just knowing it *does* work.

Reply:
The snap program is run by the snapd runner/daemon/whatever. A snap does not live in its container, but more or less has its permissions restricted by apparmor (sort of "access nothing by default"). Its plugs in the app declaration allow access to certain paths (r, rw, rwx. Whatever is appropriate) or sockets thereafter.

So /home/peter/myfile.txt is the same for every snap, but not every snap has permissions to access it. Some paths however behave like you would expect in a container. In our case the layout definition in snapcraft.yaml effects an overlay mount in the snap's mount-namespace .